### ISSUE TITLE
Load Banlance cannot start on docker 17.04.0-ce

### ISSUE THREAD

**Issue by zhangalex** [rancher/rancher#8509]
Load Banlance cannot start on docker 17.04.0-ce

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.5.5` Cattle: `v0.178.3` UI: `v1.5.8` |
|Access|`github` `admin`|
|Orchestration|`Cattle`|
|Route|`stack.index`|

**Comment by zhangalex** [comment]
It's been fixed in v1.5.6.

Thank you!

**Comment by zhangalex** [comment]
It broke again after few minutes later. The logs detail is : 

`goroutine 33 [chan receive]:
4/27/2017 3:01:54 PMmain.handleSigterm(0x7ffbe7104b20, 0xc8200c9260, 0x7ffbe7104bc0, 0xc820257b00)
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/main.go:55 +0x16d
4/27/2017 3:01:54 PMcreated by main.main
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/main.go:45 +0x29c
4/27/2017 3:01:54 PM
4/27/2017 3:01:54 PMgoroutine 12 [select]:
4/27/2017 3:01:54 PMnet/http.(*persistConn).writeLoop(0xc8202e20d0)
4/27/2017 3:01:54 PM	/usr/local/go/src/net/http/transport.go:1277 +0x472
4/27/2017 3:01:54 PMcreated by net/http.(*Transport).dialConn
4/27/2017 3:01:54 PM	/usr/local/go/src/net/http/transport.go:858 +0x10cb
4/27/2017 3:01:54 PM
4/27/2017 3:01:54 PMgoroutine 13 [select, locked to thread]:
4/27/2017 3:01:54 PMruntime.gopark(0x296c1f0, 0xc8202d9728, 0x2555fd0, 0x6, 0x100000000000018, 0x2)
4/27/2017 3:01:54 PM	/usr/local/go/src/runtime/proc.go:262 +0x163
4/27/2017 3:01:54 PMruntime.selectgoImpl(0xc8202d9728, 0x0, 0x18)
4/27/2017 3:01:54 PM	/usr/local/go/src/runtime/select.go:392 +0xa67
4/27/2017 3:01:54 PMruntime.selectgo(0xc8202d9728)
4/27/2017 3:01:54 PM	/usr/local/go/src/runtime/select.go:215 +0x12
4/27/2017 3:01:54 PMruntime.ensureSigM.func1()
4/27/2017 3:01:54 PM	/usr/local/go/src/runtime/signal1_unix.go:279 +0x358
4/27/2017 3:01:54 PMruntime.goexit()
4/27/2017 3:01:54 PM	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1
4/27/2017 3:01:54 PM
4/27/2017 3:01:54 PMgoroutine 35 [semacquire]:
4/27/2017 3:01:54 PMsync.runtime_Syncsemacquire(0xc820428690)
4/27/2017 3:01:54 PM	/usr/local/go/src/runtime/sema.go:241 +0x201
4/27/2017 3:01:54 PMsync.(*Cond).Wait(0xc820428680)
4/27/2017 3:01:54 PM	/usr/local/go/src/sync/cond.go:63 +0x9b
4/27/2017 3:01:54 PMgithub.com/rancher/lb-controller/vendor/k8s.io/kubernetes/pkg/util/workqueue.(*Type).Get(0xc8200c42d0, 0x0, 0x0, 0x0)
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/vendor/k8s.io/kubernetes/pkg/util/workqueue/queue.go:124 +0x102
4/27/2017 3:01:54 PMgithub.com/rancher/lb-controller/utils.(*TaskQueue).worker(0xc820257aa0)
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/utils/utils.go:58 +0x30
4/27/2017 3:01:54 PMgithub.com/rancher/lb-controller/utils.(*TaskQueue).(github.com/rancher/lb-controller/utils.worker)-fm()
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/utils/utils.go:33 +0x20
4/27/2017 3:01:54 PMgithub.com/rancher/lb-controller/vendor/k8s.io/kubernetes/pkg/util/wait.JitterUntil.func1(0xc820027f90)
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/vendor/k8s.io/kubernetes/pkg/util/wait/wait.go:84 +0x19
4/27/2017 3:01:54 PMgithub.com/rancher/lb-controller/vendor/k8s.io/kubernetes/pkg/util/wait.JitterUntil(0xc820027f90, 0x3b9aca00, 0x0, 0x1, 0xc8200c9200)
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/vendor/k8s.io/kubernetes/pkg/util/wait/wait.go:85 +0xb4
4/27/2017 3:01:54 PMgithub.com/rancher/lb-controller/vendor/k8s.io/kubernetes/pkg/util/wait.Until(0xc820027f90, 0x3b9aca00, 0xc8200c9200)
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/vendor/k8s.io/kubernetes/pkg/util/wait/wait.go:47 +0x43
4/27/2017 3:01:54 PMgithub.com/rancher/lb-controller/utils.(*TaskQueue).Run(0xc820257aa0, 0x3b9aca00, 0xc8200c9200)
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/utils/utils.go:33 +0x48
4/27/2017 3:01:54 PMcreated by github.com/rancher/lb-controller/controller/rancher.(*LoadBalancerController).Run
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/controller/rancher/rancher.go:106 +0x19c
4/27/2017 3:01:54 PM
4/27/2017 3:01:54 PMgoroutine 36 [runnable]:
4/27/2017 3:01:54 PMsyscall.Syscall(0x0, 0x9, 0xc82003a910, 0x8, 0x0, 0x8, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/syscall/asm_linux_amd64.s:18 +0x5
4/27/2017 3:01:54 PMsyscall.readlen(0x9, 0xc82003a910, 0x8, 0x4, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/syscall/zsyscall_linux_amd64.go:1085 +0x51
4/27/2017 3:01:54 PMsyscall.forkExec(0xc82052a259, 0x7, 0xc82052c360, 0x3, 0x3, 0xc82003aae0, 0x10, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/syscall/exec_unix.go:202 +0x567
4/27/2017 3:01:54 PMsyscall.StartProcess(0xc82052a259, 0x7, 0xc82052c360, 0x3, 0x3, 0xc82003aae0, 0x2, 0x4, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/syscall/exec_unix.go:240 +0x6b
4/27/2017 3:01:54 PMos.startProcess(0xc82052a259, 0x7, 0xc82052c360, 0x3, 0x3, 0xc82003ad18, 0xc82054a000, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/os/exec_posix.go:45 +0x33e
4/27/2017 3:01:54 PMos.StartProcess(0xc82052a259, 0x7, 0xc82052c360, 0x3, 0x3, 0xc82003ad18, 0x0, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/os/doc.go:28 +0x69
4/27/2017 3:01:54 PMos/exec.(*Cmd).Start(0xc8200d4280, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/os/exec/exec.go:332 +0x761
4/27/2017 3:01:54 PMos/exec.(*Cmd).Run(0xc8200d4280, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/os/exec/exec.go:259 +0x2d
4/27/2017 3:01:54 PMos/exec.(*Cmd).CombinedOutput(0xc8200d4280, 0x0, 0x0, 0x0, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/os/exec/exec.go:453 +0x310
4/27/2017 3:01:54 PMgithub.com/rancher/lb-controller/provider/haproxy.(*haproxyConfig).start(0xc8200c9560, 0x0, 0x0)
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/provider/haproxy/haproxy.go:158 +0x114
4/27/2017 3:01:54 PMgithub.com/rancher/lb-controller/provider/haproxy.(*Provider).StartHaproxy(0xc820257b00, 0x0, 0x0)
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/provider/haproxy/haproxy.go:182 +0x30
4/27/2017 3:01:54 PMgithub.com/rancher/lb-controller/provider/haproxy.(*Provider).Run(0xc820257b00, 0x0)
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/provider/haproxy/haproxy.go:190 +0x21
4/27/2017 3:01:54 PMcreated by github.com/rancher/lb-controller/controller/rancher.(*LoadBalancerController).Run
4/27/2017 3:01:54 PM	/go/src/github.com/rancher/lb-controller/controller/rancher/rancher.go:108 +0x1f3
4/27/2017 3:01:54 PM
4/27/2017 3:01:54 PMgoroutine 37 [runnable]:
4/27/2017 3:01:54 PMnet/http.setRequestCancel.func4(0x0, 0xc820526060, 0xc82051a3c0, 0xc820501740)
4/27/2017 3:01:54 PM	/usr/local/go/src/net/http/client.go:332
4/27/2017 3:01:54 PMcreated by net/http.setRequestCancel
4/27/2017 3:01:54 PM	/usr/local/go/src/net/http/client.go:341 +0x2f7
4/27/2017 3:01:54 PM
4/27/2017 3:01:54 PMgoroutine 38 [select]:
4/27/2017 3:01:54 PMnet.lookupIPDeadline(0xc820517320, 0x10, 0xed0938e80, 0x360a99d0, 0x3668ba0, 0x0, 0x0, 0x0, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/net/lookup.go:109 +0x6a6
4/27/2017 3:01:54 PMnet.internetAddrList(0x25569d0, 0x3, 0xc820517320, 0x13, 0xed0938e80, 0x360a99d0, 0x3668ba0, 0x0, 0x0, 0x0, ...)
4/27/2017 3:01:54 PM	/usr/local/go/src/net/ipsock.go:252 +0x6ee
4/27/2017 3:01:54 PMnet.resolveAddrList(0x2551dc0, 0x4, 0x25569d0, 0x3, 0xc820517320, 0x13, 0xed0938e80, 0x656d6e6f360a99d0, 0x3668ba0, 0x0, ...)
4/27/2017 3:01:54 PM	/usr/local/go/src/net/dial.go:158 +0x466
4/27/2017 3:01:54 PMnet.(*Dialer).Dial(0xc82000a640, 0x25569d0, 0x3, 0xc820517320, 0x13, 0x0, 0x0, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/net/dial.go:216 +0x124
4/27/2017 3:01:54 PMnet.(*Dialer).Dial-fm(0x25569d0, 0x3, 0xc820517320, 0x13, 0x0, 0x0, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/net/http/transport.go:38 +0x6e
4/27/2017 3:01:54 PMnet/http.(*Transport).dial(0xc8200bc000, 0x25569d0, 0x3, 0xc820517320, 0x13, 0x0, 0x0, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/net/http/transport.go:659 +0x96
4/27/2017 3:01:54 PMnet/http.(*Transport).dialConn(0xc8200bc000, 0x0, 0xc820505fb0, 0x4, 0xc820517320, 0x13, 0x0, 0x0, 0x0)
4/27/2017 3:01:54 PM	/usr/local/go/src/net/http/transport.go:762 +0x1e3d
4/27/2017 3:01:54 PMnet/http.(*Transport).getConn.func4(0xc8200bc000, 0x0, 0xc820505fb0, 0x4, 0xc820517320, 0x13, 0xc820501800)
4/27/2017 3:01:54 PM	/usr/local/go/src/net/http/transport.go:706 +0x66
4/27/2017 3:01:54 PMcreated by net/http.(*Transport).getConn
4/27/2017 3:01:54 PM	/usr/local/go/src/net/http/transport.go:708 +0x262`

**Comment by pmalsenz** [comment]
I'm getting the same error when the Galera Load Balancer tries to start up.

System
Version
------|---------
RancherOS | v1.0.1
Kernel | 4.9.24
Docker | 17.03.1-ce
Rancher Server | v1.5.5

The messages from the container log are attached here:
[galera-lb-log.txt](https://github.com/rancher/rancher/files/971823/galera-lb-log.txt)

I'll perform an update to the latest Rancher Server version this evening and report back.

**Comment by aemneina** [comment]
@zhangalex can you provide host OS details?

**Comment by pmalsenz** [comment]
Updating to the latest RancherOS and Rancher Server versions didn't solve the problem for me (Update was performed Tuesday/Wednesday last week).

**Comment by superseb** [comment]
Does this also occur when you add a loadbalancer to the stack in the UI @pmalsenz ?

**Comment by zhangalex** [comment]
@aemneina 

The host OS is centOS 7. The linux detail is 

`Linux iZuf63lb4s91rgkafxbmodZ 3.10.0-514.10.2.el7.x86_64 #1 SMP Fri Mar 3 00:04:05 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux`

Docker version 
`Docker version 17.04.0-ce, build 4845c56`

**Comment by pmalsenz** [comment]
@superseb  Sorry, haven't got time to test it. I'll report back till end of this week.

**Comment by pmalsenz** [comment]
@superseb 

I upgraded my Rancher Server a while before testing this again to v1.6.0, RancherOS to v1.0.1 (Docker is still 17.03).

I've set up a stack with a single MariaDB server yesterday and added a load balancer. Works out of the box.
Then I tested the MariaDB Galera Cluster from the Community Catalog again. This time it was working too.
I've added two more DB servers for testing. The containers are linked to the lb and are working, no problems till now.
The logs are clean, only status and config reload messages.

Seems to be solved with one of the updates (think a solution came with the update of Rancher Server to v1.0.1).


Keep up the great work. Thanks !
